File: packages/etebase_flutter/ios/Classes/EtebaseFlutterPlugin.m
#import "EtebaseFlutterPlugin.h"
#if __has_include(<etebase_flutter/etebase_flutter-Swift.h>)
#import <etebase_flutter/etebase_flutter-Swift.h>
#else
// Support project import fallback if the generated compatibility header
// is not copied when this plugin is created as a library.
// https://forums.swift.org/t/swift-static-libraries-dont-copy-generated-objective-c-header/19816
#import "etebase_flutter-Swift.h"
#endif

@implementation EtebaseFlutterPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftEtebaseFlutterPlugin registerWithRegistrar:registrar];
}
@end

File: packages/etebase_flutter/windows/etebase_flutter_plugin.cpp
#include "etebase_flutter_plugin.h"

#include <flutter/plugin_registrar_windows.h>

namespace etebase_flutter {

// static
void EtebaseFlutterPlugin::RegisterWithRegistrar(
    flutter::PluginRegistrarWindows *registrar) {
  auto plugin = std::make_unique<EtebaseFlutterPlugin>();
  registrar->AddPlugin(std::move(plugin));
}

EtebaseFlutterPlugin::EtebaseFlutterPlugin() {}

EtebaseFlutterPlugin::~EtebaseFlutterPlugin() {}

}  // namespace etebase_flutter
